### PR TITLE
Return model run status to front end

### PR DESCRIPTION
**Pull request by r-ash**
This PR will

* Get the progress from the queue and return it to the front end as part of the `model/status` endpoint

### CODE REVIEW

File: R/endpoints.R [hunk 1]
```diff
@@ -364,6 +361,21 @@ api_root <- function() {
...
+prepare_status_response <- function(value, id) {
+  set_scalar <- function(x) {
+    if (length(names(x)) > 1) {
+      lapply(x, set_scalar)
+    } else {
+      scalar(x)
+    }
+  }
+
+  response_value <- lapply(value[-which(names(value) == "progress")], scalar)
```
Comment (richfitz) [inline R/endpoints.R]:
```
lapply(value[names(value) != "progress"], scalar)
```
perhaps?

File: inst/schema/ProgressPhase.schema.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (richfitz) [inline inst/schema/ProgressPhase.schema.json]: we should think about where to put the schema for this if we're going to use it elsewhere more commonly - this one corresponds to what we discussed on slack?

Comment (r-ash) [reply inst/schema/ProgressPhase.schema.json]: Yeah this is the general one. Expecting a standard Progress message to be an array of these.

File: R/endpoints.R (hunk 1 above)
Comment (r-ash) [reply R/endpoints.R]: Much nicer thank you. I thought what I wrote was painfully complicated for what I was trying to achieve. I think I just had a confused moment and forgot `!=` existed.

**Pull request merged**